src/hooks/__tests__/useSourceApi.test.ts

```diff
@@ -0,0 +1,310 @@
+    ]);
+
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('callbackError');
+  });
+});
+
+describe('useAddSourceApi', () => {
+  let mockOnAddAlert;
+  let hookResult;
+
+  beforeEach(() => {
+    mockOnAddAlert = jest.fn();
+    const hook = renderHook(() => useAddSourceApi(mockOnAddAlert));
+    hookResult = hook?.result?.current;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should attempt an api call to add sources', () => {
+    const { apiCall } = hookResult;
+    const spyAxios = jest.spyOn(axios, 'post');
+
+    apiCall({ name: 'Lorem' });
+    expect(spyAxios.mock.calls).toMatchSnapshot('apiCall');
+  });
+
+  it('should handle success while attempting to add a source', async () => {
+    const { addSources } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() =>
+      Promise.resolve({
+        data: { name: 'Lorem', id: '123' }
+      })
+    );
+
+    await addSources({ name: 'Lorem' });
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('addSources, success');
+  });
+
+  it('should handle errors while attempting to add a source', async () => {
+    const { addSources } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() =>
+      Promise.reject(
+        Object.assign(new Error('Mock error'), {
+          isAxiosError: true,
+          response: {
+            data: {
+              message: 'Mock error'
+            }
+          }
+        })
+      )
+    );
+
+    await addSources({ name: 'Lorem' });
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('addSources, error');
+  });
+});
+
+describe('useEditSourceApi', () => {
+  let mockOnAddAlert;
+  let hookResult;
+
+  beforeEach(() => {
+    mockOnAddAlert = jest.fn();
+    const hook = renderHook(() => useEditSourceApi(mockOnAddAlert));
+    hookResult = hook?.result?.current;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should attempt an api call to edit sources', () => {
+    const { apiCall } = hookResult;
+    const spyAxios = jest.spyOn(axios, 'put');
+
+    apiCall({ id: 123, name: 'Lorem' });
+    expect(spyAxios.mock.calls).toMatchSnapshot('apiCall');
+  });
+
+  it('should handle success while attempting to edit a source', async () => {
+    const { editSources } = hookResult;
+    jest.spyOn(axios, 'put').mockImplementation(() =>
+      Promise.resolve({
+        data: { name: 'Lorem', id: '123' }
+      })
+    );
+
+    await editSources({ name: 'Lorem', id: '123' });
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('editSources, success');
+  });
+
+  it('should handle errors while attempting to edit a source', async () => {
+    const { editSources } = hookResult;
+    jest.spyOn(axios, 'put').mockImplementation(() =>
+      Promise.reject(
+        Object.assign(new Error('Mock error'), {
+          isAxiosError: true,
+          response: {
+            data: {
+              message: 'Mock error'
+            }
+          }
+        })
+      )
+    );
```

Comment:
required: same comment before about over-engineering our tests, remove the `Object.assign` in favor of a simplistic object